### PR TITLE
Autograde tests

### PR DESCRIPTION
**Pull request by nkorinek**
Added pytest tests for the autograde module!

### CODE REVIEW

**Comment by nkorinek** [comment]
@lwasser so this is a weird bug that's causing this to fail. For some reason, the tests I wrote for the `autograde` wrapper are causing tests that rely on the pytest fixture `pt_hist` in the file `test_base_data` to fail. It is giving it 106 bins instead of 6 bins. In my new tests, I never touch that fixture or that file at all. If I rename my file so that it is run after `test_base_data`, everything passes. But there should be a more elegant solution than that, right? Any idea what's going on here? Thanks!

**Comment by nkorinek** [comment]
@lwasser fixed all formatting issues and this is ready for review!

File: matplotcheck/tests/test_autograde.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        strings_expected=["Plot"],
+    )
+    assert result["description"] == "assert_title_contains"
+    assert result["pass"]
+    assert result["message"] == "default correct"
+    assert result["points"] == 2
+
+
+def test_autograde_runs_assert_fail(pt_scatter_plt):
+    """Test that a normal test is run with autograder and fails"""
```
Comment (lwasser) [inline matplotcheck/tests/test_autograde.py]:
```suggestion
    """Test that a when a test fails, an assertion error is returned"""
```

File: matplotcheck/tests/test_autograde.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    )
+    assert result["description"] == "assert_title_contains"
+    assert not result["pass"]
+    assert result["message"] == "default error"
+    assert result["points"] == 0
+    assert isinstance(result["traceback"], AssertionError)
+
+
+def test_autograde_runs_assert_pass_custom_message(pt_scatter_plt):
+    """Test that a custom message gets passed into a passed test"""
```
Comment (lwasser) [inline matplotcheck/tests/test_autograde.py]:
```suggestion
    """Test that when a custom message is provided, it is correctly returned to the user."""
```

File: matplotcheck/tests/test_autograde.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        pt_scatter_plt.assert_title_contains,
+        points=2,
+        strings_expected=["Plot"],
+        correct_message="This is correct!",
+    )
+    assert result["message"] == "This is correct!"
+
+
+def test_autograde_runs_assert_fail_custom_message(pt_scatter_plt):
+    """Test that a custom message gets passed into a failed test"""
```
Comment (lwasser) [inline matplotcheck/tests/test_autograde.py]:
```suggestion
    """Test that a custom message for a failed test gets returned"""
```

File: matplotcheck/tests/test_autograde.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        pt_scatter_plt.assert_title_contains,
+        points=2,
+        strings_expected=["NotAWord"],
+        error_message="This is wrong!",
+    )
+    assert result["message"] == "This is wrong!"
+
+
+def test_output_results_pass(pt_scatter_plt, capsys):
+    """Test that the output returns the correct points and output"""
```
Comment (lwasser) [inline matplotcheck/tests/test_autograde.py]:
```suggestion
    """Test that the correct number of points are returned when a test passes"""
```

File: matplotcheck/tests/test_autograde.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    )
+    assert 2 == ag.output_results([result])
+    assert (
+        "Results for test 'assert_title_contains':\n"
+        " Pass! default correct (2 points)\n" == capsys.readouterr().out
+    )
+
+
+def test_output_results_fail(pt_scatter_plt, capsys):
+    """Test that the output returns the correct points and output"""
```
Comment (lwasser) [inline matplotcheck/tests/test_autograde.py]:
```suggestion
    """Test that the correct points are returned when a test fails"""
```

**Review by lwasser — CHANGES_REQUESTED** [review]
looking good @nkorinek a few more changes!

**Comment by nkorinek** [comment]
@lwasser all green on CI

**Review by lwasser — APPROVED** [review]
nice work @nkorinek !!!

**Comment by lwasser** [comment]
this addressed issue #105

**Pull request merged**